vcd/metadata.go

```diff
@@ -2,18 +2,163 @@ package vcd
 
 import (
 	"fmt"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
 	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"github.com/vmware/go-vcloud-director/v2/util"
+	"regexp"
+	"strings"
 )
 
+var (
+	// IgnoreMetadataChangesConflictResolution can hold values "error", "warn" or "none", and tells Terraform whether it should
+	// give an error or just a warning if any Metadata Entry configured in HCL is affected by the 'ignore_metadata_changes'
+	// configuration.
+	// The keys of this map are obtained with ignoreMetadataChangesConflictResolutionHash.
+	IgnoreMetadataChangesConflictResolution map[string]string
+)
+
+// ignoreMetadataSchema returns the schema associated to ignore_metadata_changes for the provider configuration.
+func ignoreMetadataSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeSet,
+		Optional:    true,
+		Description: "Defines a set of `metadata_entry` that need to be ignored by this provider. All filters on this attribute are computed with a logical AND",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"resource_type": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Ignores metadata from the specific resource type",
+					ValidateDiagFunc: func(i interface{}, path cty.Path) diag.Diagnostics {
```

Comment:
Is it possible to extract this function and put it in validate_funcs.go with some comments?

Reply:
Done, thanks!